storage/storage_transport.go

```diff
@@ -0,0 +1,86 @@
+package storage
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/storage"
+)
+
+var (
+	// Transport is an ImageTransport that uses a default storage.Store or
+	// one that's it's explicitly told to use.
+	Transport StoreTransport = &storageTransport{}
+	// ErrInvalidReference is returned when ParseReference() is passed an
+	// empty reference.
+	ErrInvalidReference = errors.New("invalid reference")
+	idRegexp            = regexp.MustCompile("^(sha256:)?([0-9a-fA-F]{64})$")
+)
+
+// StoreTransport is an ImageTransport that uses a storage.Store to parse
+// references, either its own or one that it's told to use.
+type StoreTransport interface {
+	types.ImageTransport
+	SetStore(storage.Store)
+	ParseStoreReference(store storage.Store, reference string) (types.ImageReference, error)
+}
+
+type storageTransport struct {
+	store storage.Store
+}
+
+func (s *storageTransport) Name() string {
+	return "storage"
```

Comment:
When we expect skopeo and ocid and everyone else to work together, they'd be expected to default to using the same location.  There are options in ocid's configuration file that let it be pointed elsewhere, but it wasn't something that I'm told is generally expected to be changed.  But you're right that `StringWithTransport()` is ambiguous since it doesn't provide any way to resolve confusion about that.  I'm reworking the `StringWithTransport()`/`ParseReference()` combination to prefix the ID and/or name with information about which Store is being used.